plenum/common/txn_util.py

```diff
@@ -230,7 +230,10 @@ def transform_to_new_format(txn, seq_no):
     txn_time = t.pop(TXN_TIME, None)
     txn = do_req_to_txn(req_data=t,
                         req_op=t)
-    append_txn_metadata(txn, seq_no=seq_no, txn_time=txn_time)
+    append_txn_metadata(txn,
+                        seq_no=seq_no,
+                        txn_time=txn_time,
+                        txn_id=t.get(TXN_ID, None))
```

Comment:
TXN_ID needs to be popped from the txn's data, doens't it?

Reply:
No, txnId is placed on the top level of txn